### PR TITLE
Add tests cont_link_flap, system_health, platform_info, reboot to run on T2 topology for VoQ chassis

### PR DESCRIPTION
**Pull request by rawal01**
-->
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ x] Test case(new/improvement)

#### What is the motivation for this PR?
Change following tests  tests/platform_tests/link_flap/test_cont_link_flap.py, tests/platform_tests/test_platform_info.py,
 tests/platform_tests/test_reboot.py, tests/system_health/test_system_health.py on T2 chassis replacing rand_one_dut_hostname with enum_rand_one_per_hwsku_frontend_hostname 
#### How did you do it?
 ```tests/platform_tests/conftest.py```
	change bringup_all_interfaces to use enum_rand_one_per_hwsku_frontend_hostname instead of      rand_one_dut_hostname
	modify xcvr_skip_list to generate dictionary instead of list to accomodate mulitple duts by storing port list with dut 
	hostnames as key
	
changed following platform utils to use new xcvr_skip_list returned dictionary 
    ```tests/common/platform/interface_utils.py```
        Changed xcvr_skip_list to xcvr_skip_list with dut hostname as key
    ```tests/common/platform/transceiver_utils.py```
       Changed xcvr_skip_list to xcvr_skip_list with dut hostname as key 

following tests modified to use enum per hwsku enumeration
   ```tests/common/plugins/pdu_controller/__init__.py```
      enum_rand_one_per_hwsku_frontend_hostname instead of rand_one_dut_hostname
  
  ```tests/platform_tests/link_flap/test_cont_link_flap.py```
      enum_rand_one_per_hwsku_frontend_hostname instead of rand_one_dut_hostname
    
  ```tests/platform_tests/test_platform_info.py```
     enum_rand_one_per_hwsku_frontend_hostname instead of rand_one_dut_hostname
  
  ```tests/platform_tests/test_reboot.py```
    enum_rand_one_per_hwsku_frontend_hostname instead of rand_one_dut_hostname
  
  ```tests/platform_tests/thermal_control_test_helper.py```
      enum_rand_one_per_hwsku_frontend_hostname instead of rand_one_dut_hostname
  
  ```tests/system_health/test_system_health.py```
      enum_rand_one_per_hwsku_frontend_hostname instead of rand_one_dut_hostname

#### How did you verify/test it?
Validated the modified tests against a chassis and pizza box

### CODE REVIEW

**Comment by rawal01** [comment]
@yxieca can you please restart the pipeline the test test_snmp_cpu failed because of loganalyzer error for syncd failure not running this test has not been changed in this PR and not related to my changes

**Comment by shubav** [comment]
@jleveque  the pipeline checks have passed. Could you kindly review this?

**Comment by anshuv-mfst** [comment]
@jleveque - could you please take a look, thanks.

File: tests/common/platform/transceiver_utils.py [hunk 1]
```diff
@@ -45,7 +45,8 @@ def all_transceivers_detected(dut, asic_index, interfaces, xcvr_skip_list):
     asichost = dut.asic_instance(asic_index)
     docker_cmd = asichost.get_docker_cmd(cmd, "database")
     db_output = dut.command(docker_cmd)["stdout_lines"]
-    not_detected_interfaces = [intf for intf in interfaces if "TRANSCEIVER_INFO|%s" % intf not in db_output]
+    not_detected_interfaces = [intf for intf in interfaces if intf not in xcvr_skip_list[dut.hostname] if
+                               "TRANSCEIVER_INFO|%s" % intf not in db_output]
```
Comment (jleveque) [inline tests/common/platform/transceiver_utils.py:49]: Suggest refactoring for readability:

```python
    not_detected_interfaces = [intf for intf in interfaces if (intf not in xcvr_skip_list[dut.hostname] and
                                                               "TRANSCEIVER_INFO|{}".format(intf) not in db_output)]
```

File: tests/common/plugins/pdu_controller/__init__.py [hunk 2]
```diff
@@ -8,16 +8,21 @@
...
     @returns: Returns a pdu controller object implementing the BasePduController interface defined in
               controller_base.py.
     """
-    duthost = duthosts[rand_one_dut_hostname]
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
     inv_mgr = duthost.host.options["inventory_manager"]
     pdu_host_list = inv_mgr.get_host(duthost.hostname).get_vars().get("pdu_host")
+    if not pdu_host_list:
+        logging.info("No 'pdu_host' is defined in inventory file for '%s'. Unable to create pdu_controller" %
+                     duthost.hostname)
```
Comment (jleveque) [inline tests/common/plugins/pdu_controller/__init__.py:23]: Suggest using more modern `.format()` rather than `%` operator.

File: tests/platform_tests/test_reboot.py [hunk 3]
```diff
@@ -81,20 +81,24 @@ def check_interfaces_and_services(dut, interfaces, xcvr_skip_list, reboot_type =
...
-    logging.info("Wait %d seconds for all the transceivers to be detected" % MAX_WAIT_TIME_FOR_INTERFACES)
-    assert wait_until(MAX_WAIT_TIME_FOR_INTERFACES, 20, check_all_interface_information, dut, interfaces, xcvr_skip_list), \
-        "Not all transceivers are detected or interfaces are up in %d seconds" % MAX_WAIT_TIME_FOR_INTERFACES
+    if dut.is_supervisor_node():
+        logging.info("skipping interfaces related check for supervisor")
+    else:
+        logging.info("Wait %d seconds for all the transceivers to be detected" % MAX_WAIT_TIME_FOR_INTERFACES)
+        assert wait_until(MAX_WAIT_TIME_FOR_INTERFACES, 20, check_all_interface_information, dut, interfaces,
+                          xcvr_skip_list), \
+            "Not all transceivers are detected or interfaces are up in %d seconds" % MAX_WAIT_TIME_FOR_INTERFACES
```
Comment (jleveque) [inline tests/platform_tests/test_reboot.py:90]: I prefer eliminating the backslash. Maybe something like:

```python
        result = wait_until(MAX_WAIT_TIME_FOR_INTERFACES, 20, check_all_interface_information, dut, interfaces, xcvr_skip_list)
        assert result, "Not all transceivers are detected or interfaces are up in {} seconds".format(MAX_WAIT_TIME_FOR_INTERFACES)
```

**Review by jleveque — CHANGES_REQUESTED** [review]

**Comment by shubav** [comment]
@jleveque how do we trigger a review by sonic-fundamentals? Can you help? Thanks.

**Comment by shubav** [comment]
> @jleveque how do we trigger a review by sonic-fundamentals? Can you help? Thanks.

@anshuv-mfst , could you please trigger a review by sonic=-fundamentals? Joe has tagged them, but could you please help close this? Thanks.

**Comment by anshuv-mfst** [comment]
Hi @yxieca - could you please assign this PR, thanks.

File: tests/common/plugins/pdu_controller/__init__.py [hunk 4]
```diff
@@ -18,6 +18,11 @@ def pdu_controller(duthosts, enum_rand_one_per_hwsku_hostname, conn_graph_facts,
     duthost = duthosts[enum_rand_one_per_hwsku_hostname]
     inv_mgr = duthost.host.options["inventory_manager"]
     pdu_host_list = inv_mgr.get_host(duthost.hostname).get_vars().get("pdu_host")
+    if not pdu_host_list:
+        logging.info("No 'pdu_host' is defined in inventory file for {}. Unable to create pdu_controller".format(
+            duthost.hostname))
+        yield None
+        return
```
Comment (yxieca) [inline tests/common/plugins/pdu_controller/__init__.py:25]: This is a good change but doesn't seem to belong to this PR? Can you separate this change in a different PR?

**Review by yxieca — COMMENTED** [review]
Please separate the PDU change to another PR. The rest looks good to me. The PDU change also looks good but it should belong to a separate PR.

File: tests/common/plugins/pdu_controller/__init__.py (hunk 4 above)
Comment (rawal01) [reply tests/common/plugins/pdu_controller/__init__.py]: @yxieca I pulled that changes and will create separate PR, please re-review

**Review by yxieca — APPROVED** [review]

**Comment by yxieca** [comment]
@jleveque can you review this change again?

**Review by jleveque — APPROVED** [review]

**Comment by anshuv-mfst** [comment]
Hi @yxieca - could you please merge the PR, thanks.
